src/main/java/com/capgemini/wsb/fitnesstracker/training/internal/TrainingController.java

```diff
@@ -0,0 +1,86 @@
+package com.capgemini.wsb.fitnesstracker.training.internal;
+
+import com.capgemini.wsb.fitnesstracker.training.api.Training;
+import com.capgemini.wsb.fitnesstracker.training.api.TrainingNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+@RestController
+@RequestMapping("/v1/trainings")
+@RequiredArgsConstructor
+class TrainingController {
+
+    private final TrainingServiceImpl trainingService;
+    private final TrainingMapper trainingMapper;
+
+    @GetMapping
+    public List<TrainingDto> getAllTrainings() {
+        return trainingService.findAllTrainings().stream()
+                .map(trainingMapper::toDto)
+                .toList();
+    }
+
+    @GetMapping("/{userId}")
+    public List<TrainingDto> getTrainingsForUser(@PathVariable Long userId) {
+        return trainingService.findTrainingsByUserId(userId).stream()
+                .map(trainingMapper::toDto)
+                .toList();
+    }
+
+    @GetMapping("/finished/{afterTime}")
+    public List<TrainingDto> getTrainingsFinishedAfter(
+            @PathVariable String afterTime) {
+        try {
+            SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+            Date date = format.parse(afterTime);
+            return trainingService.findTrainingsFinishedAfter(date).stream()
+                    .map(trainingMapper::toDto)
+                    .toList();
+        } catch (ParseException e) {
+            throw new ResponseStatusException(
+                    HttpStatus.BAD_REQUEST,
+                    "Invalid date format. Use yyyy-MM-dd"
+            );
+        }
+    }
+
+    @GetMapping("/activityType")
+    public List<TrainingDto> getTrainingsByActivityType(
+            @RequestParam String activityType) {
+        if (activityType == null) {
+            throw new ResponseStatusException(
+                    HttpStatus.BAD_REQUEST,
+                    "Activity type must be provided"
+            );
+        }
+        return trainingService.findTrainingsByActivityType(activityType).stream()
+                .map(trainingMapper::toDto)
+                .toList();
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public TrainingDto createTraining(@RequestBody TrainingCreateDto trainingDto) {
+        Training training = trainingMapper.toEntity(trainingDto);
+        Training createdTraining = trainingService.createTraining(training);
+        return trainingMapper.toDto(createdTraining);
+    }
+
+    @PutMapping("/{trainingId}")
```

Comment:
Update doesnt work correctly, fields need to made nullable so it doesnt require to provide all the information even when trying to change only 1 field for example.

Reply:
![image](https://github.com/user-attachments/assets/1d88f48f-10d3-4e43-9e7b-9ac7b5931b58)

Reply:
thats why update should have its own DTO